src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1305 @@
+
+    LAPC_SNAPSHOT_STATUS_0_OFFSET, LAPC_SNAPSHOT_STATUS_1_OFFSET,
+    LAPC_SNAPSHOT_STATUS_2_OFFSET, LAPC_SNAPSHOT_STATUS_3_OFFSET
+  };
+
+  uint32_t currData[XLAPC_STATUS_PER_SLOT];
+  
+  for (int c=0; c < XLAPC_STATUS_PER_SLOT; c++) {
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    size += xclRead(handle, XCL_ADDR_SPACE_DEVICE_CHECKER, dbgIpInfo->m_base_address+statusRegisters[c], &currData[c], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+  }
+
+  lapcResults.OverallStatus[index]      = currData[XLAPC_OVERALL_STATUS];
+  std::copy(currData+XLAPC_CUMULATIVE_STATUS_0, currData+XLAPC_SNAPSHOT_STATUS_0, lapcResults.CumulativeStatus[index]);
+  std::copy(currData+XLAPC_SNAPSHOT_STATUS_0, currData+XLAPC_STATUS_PER_SLOT, lapcResults.SnapshotStatus[index]);
+
+  (void)size;
+
+  // increment debugIpNum
+  ++debugIpNum[LAPC];
+  lapcResults.NumSlots = (unsigned int)debugIpNum[LAPC];
+}
+
+void 
+DebugIpStatusCollector::printLAPCResults(std::ostream& _output)
+{ 
+  if(0 == lapcResults.NumSlots) {
+    return;
+  }
+
+  bool violations_found = false;
+  bool invalid_codes = false;
+  _output << "\nLight Weight AXI Protocol Checkers codes \n";
+  auto col1 = std::max(cuNameMaxStrLen[LAPC], strlen("CU Name")) + 4;
+  auto col2 = std::max(portNameMaxStrLen[LAPC], strlen("AXI Portname"));
+
+  for (size_t i = 0; i < lapcResults.NumSlots; ++i) {
+    if (!xclAXICheckerCodes::isValidAXICheckerCodes(lapcResults.OverallStatus[i],
+                lapcResults.SnapshotStatus[i], lapcResults.CumulativeStatus[i])) {
+      _output << "CU Name: " << cuNames[LAPC][i] << " AXI Port: " << portNames[LAPC][i] << "\n";
+      _output << "  Invalid codes read, skip decoding\n";
+      invalid_codes = true;
+    } else if (lapcResults.OverallStatus[i]) {
+      _output << "CU Name: " << cuNames[LAPC][i] << " AXI Port: " << portNames[LAPC][i] << "\n";
+      _output << "  First violation: \n";
+      _output << "    " <<  xclAXICheckerCodes::decodeAXICheckerCodes(lapcResults.SnapshotStatus[i]);
+      //snapshot reflects first violation, cumulative has all violations
+      unsigned int tCummStatus[4];
+      std::transform(lapcResults.CumulativeStatus[i], lapcResults.CumulativeStatus[i]+4, lapcResults.SnapshotStatus[i], tCummStatus, std::bit_xor<unsigned int>());
+      _output << "  Other violations: \n";
+      std::string tstr = xclAXICheckerCodes::decodeAXICheckerCodes(tCummStatus);
+      if (tstr == "") {
+        _output << "    None";
+      } else {
+        _output << "    " <<  tstr;
+      }
+      violations_found = true;
+    }
+  }
+  if (!violations_found && !invalid_codes)
+    _output << "No AXI violations found \n";
+
+  if (violations_found && /*aVerbose &&*/ !invalid_codes) {
+    std::ofstream saveFormat;
+    saveFormat.copyfmt(_output);
+
+    _output << "\n";
+    _output << std::left
```

Comment:
Yep, boost format :-)

Reply:
boost::format is really nice and easy to use and read. Thanks for your suggestion. I have added change at all the required places.